### ISSUE TITLE
gcc no longer  passing -I directories to assembler

### ISSUE THREAD

**Issue by jstraus59** [foss-for-synopsys-dwc-arc-processors/toolchain#63]
In release 2014.012 directories specified with "-I" on the  gcc command line were passed to the assembler, but on release 2015.06. I have created a simple test case to show this:

File test.S:

```
.include "test.inc"

    testMACRO
```

File include/test.inc:

```
.macro testMacro
    clri    %r16
.endm
```

The following command using the prebuilt [2014.12](https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/releases/download/arc-2014.12/arc_gnu_2014.12_prebuilt_elf32_le_linux_install.tar.gz) release  works:

```
./arc_gnu_2014.12_prebuilt_elf32_le_linux_install/bin/arc-elf32-gcc -mcpu=arcem -c -o test.o test.S -I include
```

while the same command using the prebuilt [2015.06](https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/releases/download/arc-2015.06/arc_gnu_2015.06_prebuilt_elf32_le_linux_install.tar.gz) release does not work:

```
arc_gnu_2015.06_prebuilt_elf32_le_linux_install/bin/arc-elf32-gcc -mcpu=arcem -c -o test.o test.S -I include
/tmp/cc8BGSXu.s: Assembler messages:
/tmp/cc8BGSXu.s:5: Error: can't open test.inc for reading: No such file or directory
test.S:3: Error: bad instruction `testmacro'
```

If we run both with the -v option we can see that with 2014.12 the assembler command includes "-I include" but with 2015.06 it does not include any -I directories.

Is there some reason for this change? Is there some workaround to get it to work?

Thanks

Jim Straus

**Comment by claziss** [comment]
I need to check what is going on. For the time being, you can pass directly to the assembler the include folder via -Wa driver option. For example:

```
arc-elf32-gcc -mcpu=arcem -c -o test.o test.S -I ./include -Wa,-I -Wa,include
```

**Comment by jstraus59** [comment]
That approach works as a temporary workaround. 

Thanks for the info and for looking into the issue.

Jim

**Comment by claziss** [comment]
I found the possible source of your issue. It seems the configure file doesn't recognize we have GNU assembler and GNU linker. Hence, will not pass a number of options to as/ld, including the -I option.

**Comment by claziss** [comment]
The issue is how the compiler is configured. We need to pass --with-gnu-ld and --with-gnu-as to the configure line. Assigned this to Anton.

**Comment by anthony-kolesov** [comment]
That should be fixed in a0febf3b7e5583d311673391ef617c0cb4d92dc0.

Jim, if you are building you own toolchain, you can add this to build-all.sh arguments: `--config-extra "--with-gnu-ld --with-gnu-as"`.
